File: packages/roadbook-api/models/schedule.js
'use strict';
const { Model } = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Schedule extends Model {
    static associate(models) {
      models.Travel.hasMany(this, { foreignKey: "tId" });
    }
  }
  Schedule.init({
    id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      autoIncrement: true,
      primaryKey: true,
    },
    cover: DataTypes.STRING, // 封面
    name: DataTypes.STRING, // 行程名称
    coordinate: DataTypes.STRING, //坐标经纬度【高德】
    address: DataTypes.STRING, // 地址
    dianpingUUID: DataTypes.STRING, //点评uuid
    isHotel: DataTypes.BOOLEAN, // 是否为酒店日程
    startTime: DataTypes.STRING, // 出发时间
    endTime: DataTypes.STRING, // 离店时间，当日程为酒店时有效
    traffic: DataTypes.STRING, //交通方式
    screenshots: DataTypes.STRING, // 票据截图
    notes: DataTypes.STRING // 攻略
  }, {
    sequelize,
    timestamps: false,
    modelName: 'Schedule',
  });
  return Schedule;
};

File: packages/roadbook-flutter/lib/shared/models/schedule.dart
// lib/shared/models/schedule.dart
class Schedule {
  const Schedule({
    this.id,
    required this.tId,
    required this.name,
    required this.coordinate,
    required this.address,
    this.cover,
    this.dianpingUUID,
    required this.isHotel,
    this.startTime,
    this.endTime,
    this.screenshots,
    this.notes,
  });

  final int? id;
  final int tId;
  final String name;
  final String coordinate;
  final String address;
  final String? cover;
  final String? dianpingUUID;
  final bool isHotel;
  final DateTime? startTime;
  final DateTime? endTime;
  final String? screenshots;
  final String? notes;

  List<String> get screenshotList {
    if (screenshots == null || screenshots!.isEmpty) return [];
    return screenshots!.split(',').where((s) => s.isNotEmpty).toList();
  }

  factory Schedule.fromJson(Map<String, dynamic> json) => Schedule(
        id: json['id'] as int?,
        tId: json['tId'] as int,
        name: json['name'] as String,
        coordinate: json['coordinate'] as String,
        address: json['address'] as String,
        cover: json['cover'] as String?,
        dianpingUUID: json['dianpingUUID'] as String?,
        isHotel: json['isHotel'] as bool? ?? false,
        startTime: json['startTime'] != null
            ? DateTime.parse(json['startTime'] as String)
            : null,
        endTime: json['endTime'] != null
            ? DateTime.parse(json['endTime'] as String)
            : null,
        screenshots: json['screenshots'] as String?,
        notes: json['notes'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'tId': tId,
        'name': name,
        'coordinate': coordinate,
        'address': address,
        'cover': cover,
        'dianpingUUID': dianpingUUID,
        'isHotel': isHotel,
        'startTime': startTime?.toString(),
        'endTime': endTime?.toString(),
        'screenshots': screenshots,
        'notes': notes,
      };
}
